packages/blade/src/components/Dropdown/Dropdown.tsx

```diff
@@ -0,0 +1,89 @@
+import React from 'react';
+import { DropdownContext, useDropdown } from './useDropdown';
+import type { DropdownContextType } from './useDropdown';
+import { useId } from '~src/hooks/useId';
+import Box from '~components/Box';
+import { getPlatformType } from '~utils';
+
+type DropdownProps = {
+  selectionType?: 'single' | 'multiple';
+  children: React.ReactNode[];
+};
+
+const platformType = getPlatformType();
+const isReactNative = platformType === 'react-native';
+
+/**
+ * **Dropdown component**
+ *
+ * **Usage**
+ *
+ * ```jsx
+ * <Dropdown selectionType="single">
+ *  <SelectInput />
+ *  <DropdownOverlay>
+ *    <ActionList>
+ *      <ActionListItem />
+ *      <ActionListItem />
+ *    </ActionList>
+ *  </DropdownOverlay>
+ * </Dropdown>
+ * ```
+ */
+function Dropdown({ children, selectionType = 'single' }: DropdownProps): JSX.Element {
+  const [isOpen, setIsOpen] = React.useState(false);
+  const [options, setOptions] = React.useState<DropdownContextType['options']>([]);
+  const [selectedIndices, setSelectedIndices] = React.useState<
+    DropdownContextType['selectedIndices']
+  >([]);
+  const [activeIndex, setActiveIndex] = React.useState(-1);
+  const [shouldIgnoreBlur, setShouldIgnoreBlur] = React.useState(false);
+  const selectInputRef = React.useRef<HTMLButtonElement>(null);
```

Comment:
once you rebase i think you'll have to merge the refs coming from client with this ref